### PR TITLE
Remap ports to the non-conflicting range

### PR DESCRIPTION
**Pull request by dhruvkb**
## Fixes

Addresses WordPress/openverse#200

## Description

This PR
- removes binding for ports that are only used for internal communication
- changes port mappings to ports in the 502xx range to prevent any conflicts with services running on the host
- updates documentation and tests to keep up with the changes
- fixes some small side issues along the way
  - removing unused settings
  - using env vars in the `run.sh` entrypoint code and aligning it with the `justfile`
  - using the `_api-up` recipe in more places
  - converting the NGINX conf into [a template to use env vars](https://github.com/docker-library/docs/tree/master/nginx#using-environment-variables-in-nginx-configuration-new-in-119)

## Testing Instructions

0. Check out the PR and run the dev server.
1. Visit some endpoints that can help gauge the communication between processes e.g.
   - the stats endpoint, to test ES and cache
   - the thumbnails endpoint, to test the thumbnails service

Apart from this, a passing CI is pretty good proof of inter-process communication working normally.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: api/run.sh
Comment (sarayourfriend) [inline api/run.sh:5]: Can this line be split up?

```suggestion
es_url = "http://${ELASTICSEARCH_URL:-es}:${ELASTICSEARCH_PORT:-9200}/_cluster/health"
while [ "$(curl -s -o /dev/null -w '%{http_code}' "$es_url")" != "200" ]; do
```

Would this work?

File: justfile
Comment (sarayourfriend) [inline justfile:234]: Would we want to change this port to something in 502xx as well?

Comment (dhruvkb) [reply justfile:234]: The 3000 here refers to the port of the container (running inside `just exec`). It is correctly set to 3000.

File: api/run.sh
Comment (dhruvkb) [reply api/run.sh]: It works, except for the spaces around `=` (shell scripts, right?). Fixed in 6df39aa.

File: .github/workflows/ci_cd.yml
Comment (sarayourfriend) [inline .github/workflows/ci_cd.yml:181]: It doesn't look like the ingestion server is needed for the api doctest recipe.
```suggestion

```

File: api/docs/guides/test.md
Comment (sarayourfriend) [inline api/docs/guides/test.md:23]: Why is this removed?

File: docker-compose.yml
Comment (sarayourfriend) [inline docker-compose.yml:6]: I think some folks on occasion do connect to the local DB using psql. It'd be nice to leave this one mapped to the host as well.

File: docker-compose.yml
Comment (sarayourfriend) [inline docker-compose.yml:38]: Ditto with Redis. It can be useful to be able to debug with the Redis CLI locally.

File: justfile
Comment (sarayourfriend) [inline justfile:78]: This seems kind of like a weird way to write this recipe. Wouldn't this be a little more expected/natural?

```suggestion
# Ensure all services are up and running
@_api-up:
    just up
    just wait-for-ing
    just wait-for-web
```

File: docker-compose.yml
Comment (sarayourfriend) [inline docker-compose.yml:76]: Just a small suggestion because this always confuses me when I check the port mappings in the docker compose. I never remember which to use for the Django API.

```suggestion
      # django
      - "50280:8000"
      # sphinx
      - "50230:3000"
```

**Review by sarayourfriend — CHANGES_REQUESTED** [review]
LGTM! Tested fine locally. Only one of the suggestions I'm strong-ish on is mapping the Postgres and Redis ports. Otherwise this is great.

File: api/docs/guides/test.md
Comment (dhruvkb) [reply api/docs/guides/test.md:23]: Since I hadn't mapped the DBs and Redis to the host, testing locally was not possible. Now that you and @AetherUnbound have expressed support for mapping these services locally, we can restore this functionality.

File: justfile
Comment (dhruvkb) [reply justfile]: I don't know who wrote the recipe like that but it's been fixed in 670dd8a.

File: docker-compose.yml
Comment (dhruvkb) [reply docker-compose.yml]: Added comments in faa97ae.

File: docker-compose.yml
Comment (dhruvkb) [reply docker-compose.yml]: Exposed port in d714b04.

File: docker-compose.yml
Comment (dhruvkb) [reply docker-compose.yml]: I have never done this but that shouldn't prevent anyone else from being able to do so. Exposed port in d714b04.

File: api/docs/guides/test.md
Comment (sarayourfriend) [reply api/docs/guides/test.md:23]: I typically use `api-testlocal` when running unit tests locally.

Comment (dhruvkb) [reply api/docs/guides/test.md:23]: Can we fix that in a separate PR? Our tests are fundamentally flawed in that there is no separation between unit tests which should run in the Django environment and E2E/integration tests which should run outside. This means that when ports are different  inside and outside the container they cannot use the same env vars.

Updating tests would make this PR quite big and hard to review. So I'm hoping we can open a high-priority issue for that and fix it in post.

As a workaround, you can speed up test execution by running specific tests by passing a filename after `just api-test <file_name>`.

Comment (sarayourfriend) [reply api/docs/guides/test.md:23]: I always run specific tests when writing new ones, the issue is the container startup time and difficulties with attaching debuggers.

I'm fine solving it in a follow up PR but please open an issue describing the problem and the solution. I'm not entirely convinced the integration tests need to run outside the container (see Playwright tests in the frontend, for example) but we can discuss the trade offs in another issue as you suggested.

**Review by sarayourfriend — APPROVED** [review]
LGTM!

**Comment by dhruvkb** [comment]
@obulat could you please review the PR when you have some time? Thanks!

**Comment by krysal** [comment]
@dhruvkb After Olga's review, be sure to merge this PR after the v2.5.10 is cut, please. This is looking like a non-trivial change.

**Comment by sarayourfriend** [comment]
Good point @krysal. Should just hold off on merging this at all until the ES situation is resolved?

**Comment by dhruvkb** [comment]
We did agree to no merge anything unrelated to ES but my interpretation was to not touch the ES-related bits. I can definitely hold off on merging this but this only affects the dev setup, there is no change to the code itself.

I'll wait for the v2.5.0 to be cut before merging and block it till then.

**Comment by dhruvkb** [comment]
Merging since 2.5.10 has been cut and deployed to production.

File: .github/workflows/ci_cd.yml
Comment (AetherUnbound) [inline .github/workflows/ci_cd.yml:181]: I don't know if this is possible with GHA, but it would be really cool to have this as a set of steps that we could just reference as a precursor to specific tasks! Since this does appear to be reused elsewhere in this file.

File: api/docs/guides/test.md
Comment (AetherUnbound) [reply api/docs/guides/test.md:23]: FWIW the catalog has a just recipe for setting up a "test session" `docker run`, which I use frequently for rapidly iterating on tests: 
https://github.com/WordPress/openverse-catalog/blob/40b4306b161bb4328f4b2e77bc9790e34332eb41/justfile#L65-L67

Perhaps it could be used here!

**Review by AetherUnbound — APPROVED** [review]
Looks and works great!

File: .github/workflows/ci_cd.yml
Comment (sarayourfriend) [reply .github/workflows/ci_cd.yml:181]: It is possible using YAML anchors but `actionlint` does not support them yet: https://github.com/rhysd/actionlint/issues/133

**Pull request merged**
